mimic/test/test_ironic.py

```diff
@@ -45,19 +45,114 @@ def setUp(self):
+        """
+        Create a new node and return node.
+        """
+        request = create_request or self.create_request
+        (response, content) = self.successResultOf(json_request(
+            self, self.root, "POST", self.url, body=json.dumps(request)))
+        self.assertEqual(response.code, 201)
+        return content
 
     def get_nodes(self, postfix=None):
         """
         Get nodes and return content
         """
-        url = "/ironic/v1/nodes"
+        url = self.url
         if postfix:
-            url = "/ironic/v1/nodes" + postfix
+            url = self.url + postfix
         (response, content) = self.successResultOf(json_request(
             self, self.root, "GET", url))
         self.assertEqual(200, response.code)
         return content
 
+    def test_create_node(self):
+        """
+        Create node returns 201 and the newly created node.
+        """
+        new_node = self.create_node()
+        for key in self.create_request.keys():
+            self.assertEqual(self.create_request[key], new_node[key])
+
+    def test_create_node_failure(self):
+        """
+        Create node returns 400 if the request json is not
+        as expected.
+        """
+        response = self.successResultOf(request(
+            self, self.root, "POST", self.url, body=json.dumps({})))
+        self.assertEqual(response.code, 400)
+
+    def test_delete_node_when_node_does_not_exist(self):
+        """
+        Delete node returns 404 when the given node_id
+        does not exist.
+        """
+        response = self.successResultOf(request(
+            self, self.root, "DELETE", self.url + "/1234"))
+        self.assertEqual(response.code, 404)
+
+    def test_delete_node(self):
+        """
+        Delete node returns 204 deletes the given node_id.
+        """
+        # create node
+        content = self.create_node()
+        node_id = str(content['uuid'])
+
+        # delete node
+        response = self.successResultOf(request(
+            self, self.root, "DELETE", self.url + '/' + node_id))
+        self.assertEqual(response.code, 204)
+
+        # get node
+        response = self.successResultOf(request(
+            self, self.root, "GET", self.url + '/' + node_id))
+        self.assertEqual(404, response.code)
+
+    def test_create_then_get_node(self):
+        """
+        Test create node then get the node and verify attributes
+        """
+        # create node
+        content = self.create_node({"properties": {"memory_mb": 32768}})
+        node_id = str(content['uuid'])
+
+        # get node
+        (response, get_content) = self.successResultOf(json_request(
+            self, self.root, "GET", self.url + '/' + node_id))
+        self.assertEqual(200, response.code)
+        self.assertEqual(content, get_content)
+
+    def test_create_then_get_node_2(self):
+        """
+        Test create node then get the node and verify attributes
+        """
```

Comment:
This test case is almost exactly the same as the one above it. Is it a duplicate?